### DISCUSSION TITLE
Extracting 3D Geometries

### DISCUSSION THREAD

**Discussion by Saeid-Emam** [iTwin/itwinjs-core#2969] [Q&A] [answered]
Is there any way to **extract geometries** created by codes to any format (like obj, stl, ...)? 3D geometries like the ones in this sample:
_https://www.itwinjs.org/sample-showcase/?group=Geometry+Samples&sample=advanced-3d-sample_

I would like to run a **clash detection test** between these geometries and models/categories.
Any help would be appreciated.

**Comment by bbastings** [comment, marked as answer]
The easiest approach to getting a representation of surface/solid geometry is to use [ExportGraphics](https://www.itwinjs.org/reference/imodeljs-backend/imodels/imodeldb/exportgraphics/).
